src/routing_table/mod.rs

```diff
@@ -502,6 +502,11 @@ impl<T: Binary + Clone + Copy + Debug + Default + Hash + Xorable> RoutingTable<T
     // The actual merge of the group is only done once all expected merging groups have provided
     // details.  See the docs for `OwnMergeState` for full details of the return value.
     pub fn merge_own_group(&mut self, merge_details: OwnMergeDetails<T>) -> OwnMergeState<T> {
+        if self.groups.contains_key(&merge_details.merge_prefix) {
+            warn!("{:?}: Attempt to call merge_own_group() for an already merged prefix {:?}",
+                  self.our_name.debug_binary(),
```

Comment:
For most places outside the RT tests we tend to use the [abbreviated hex notation](https://github.com/maidsafe/routing/blob/master/src/xor_name.rs#L99-L102) for `XorName`s which is also what is used in the `Debug` and `Display` traits for `XorName`, so it might be worth changing this to `warn!("{}: ...", self.our_name, ...)` so it would tie in with other log messages relating to a given node?